cmd/builder/internal/builder/templates/main.go.tmpl

```diff
@@ -38,14 +46,21 @@ func main() {
 				{{- if .ConfResolver.DefaultURIScheme }}
 				DefaultScheme: "{{ .ConfResolver.DefaultURIScheme }}",
 				{{- end }}
-				ConverterFactories: []confmap.ConverterFactory{
-					expandconverter.NewFactory(),
-				},
 			},
 		},
 		{{- end}}
 	}
 
+    {{ if .Distribution.SupportsConfmapFactories }}
+    if useUnifiedEnvVarExpansionRules.IsEnabled() && set.ConfigProviderSettings.ResolverSettings.DefaultScheme == "" {
+        set.ConfigProviderSettings.ResolverSettings.DefaultScheme = "env"
+    } else {
```

Comment:
```suggestion
    if useUnifiedEnvVarExpansionRules.IsEnabled() {
        if set.ConfigProviderSettings.ResolverSettings.DefaultScheme == "" {
            set.ConfigProviderSettings.ResolverSettings.DefaultScheme = "env"
        }
    } else {
```
Could we separate these conditions? I'd prefer if setting a default scheme didn't automatically add back the expand converter even if the flag is enabled.